script.js

```diff
@@ -1,56 +1,72 @@
-let collection = [];
-let arr = [];
+/* eslint-disable max-classes-per-file */
 
 const inputTitle = document.getElementById('title');
 const inputAuthor = document.getElementById('author');
 const submitBtn = document.querySelector('.add-btn');
 const bookSection = document.querySelector('.books');
 
-const book = {
-  title: '',
-  author: '',
-};
+class Collection {
+  constructor(books = []) {
+    this.books = books;
+  }
+
+  add(data) {
+    if (this.books.filter((item) => item.author === data.author
+    && item.title === data.title).length > 0) {
+      return;
+    }
+    this.books.push(data);
+    this.display(data);
+    this.remove();
+    this.populateStorage();
+  }
+
+  remove() {
+    const removeBtns = document.querySelectorAll('.remove-button');
+    removeBtns[removeBtns.length - 1].addEventListener('click', (evt) => {
+      bookSection.removeChild(evt.target.parentNode);
+      this.removeFromColl(evt.target);
+    });
+  }
+
+  display(data) {
+    if (this) {
+      const div = document.createElement('div');
+      div.className = 'book-wraper';
+      div.innerHTML = `<h3>${data.title}</h3>
+                    <h3>${data.author}</h3>
+                    <button data-value="${data.title}-${data.author}" type="button" class ="remove-button">Remove</button>`;
+      bookSection.appendChild(div);
+    }
+  }
+
+  removeFromColl(data) {
+    const arr = data.getAttribute('data-value').split('-');
+    this.books = this.books.filter((item) => item.title !== arr[0] && item.author !== arr[1]);
+    this.populateStorage();
+  }
 
-function populateStorage() {
-  localStorage.setItem('Books', JSON.stringify({
-    coll: collection,
-  }));
+  populateStorage() {
+    localStorage.setItem('bookCollection', JSON.stringify({
+      bookColl: this.books,
+    }));
+  }
 }
 
-function addBook(book) {
-  const div = document.createElement('div');
-  div.className = 'book-wraper';
-  div.setAttribute('data-value', `${book.title}-${book.author}`);
-  div.innerHTML = `<h3>${book.title}</h3>
-                  <h3>${book.author}</h3>
-                  <button type="button" class ="remove-button">Remove</button>`;
-  bookSection.appendChild(div);
-  const removeBtn = document.querySelectorAll('.remove-button');
-  removeBtn[removeBtn.length - 1].addEventListener('click', (evt) => {
-    document.querySelector('.books').removeChild(evt.target.parentNode);
-    arr = evt.target.parentNode.getAttribute('data-value').split('-');
-    collection = collection.filter((item) => item.title !== arr[0] && item.author !== arr[1]);
-    populateStorage();
-  });
+class Book {
+  constructor(title, author) {
+    this.title = title;
+    this.author = author;
```

Comment:
- [ ] [OPTIONAL] You can add an id for each book as it would make it easier to delete books with matching names.

Reply:
Yes, we didn't realize and added an attribute, thanks.